tests/k8st/tests/test_bgp_advert.py

```diff
@@ -152,20 +125,46 @@ def tearDown(self):
             pass
         self.delete_and_confirm("bgp-test", "ns")
 
-    def test_bgp_advert(self):
+    def get_svc_cluster_ip(self, svc, ns):
+        return run("kubectl get svc %s -n %s -o json | jq -r .spec.clusterIP" % (svc, ns)).strip()
+
+    def do_external_curl(self, container, hostname):
+        return run("docker exec %s curl --connect-timeout 2 -m 3 %s" % (container, hostname))
+
+    def test_mainline(self):
         """
-        Test that BGP routes to services are exported over BGP
+        Runs the mainline tests for service ip advertisement
+        - Create both a Local and a Cluster type NodePort service with a single replica.
+          - assert only local and service CIDR routes are advertised.
+          - assert /32 routes are used, source IP is preserved.
+        - Create a local LoadBalancer service with clusterIP = None, assert no change.
+        - Scale the Local NP service so it is running on multiple nodes, assert ECMP routing, source IP is preserved.
+        - Delete both services, assert only cluster CIDR route is advertised.
         """
+        # Assert that a route to the service IP range is present
+        retry_until_success(lambda: self.assertIn("10.96.0.0/12", self.get_routes()), retries=5, wait_time=1)
+
+        # Create nginx deployment and service
+        self.create_service("nginx:1.7.9", "nginx", "bgp-test", 80)
+        self.create_service("nginx:1.7.9", "nginx2", "bgp-test", 80, traffic_policy="Cluster")
```

Comment:
nit: I'd call these `nginx-local` and `nginx-cluster` for easier time debugging if things go wrong.